service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -107,30 +107,56 @@ sudo cp terra /usr/bin/terra
 sudo -u "${JUPYTER_USER}" sh -c "terra config set browser MANUAL"
 # Set the CLI terra server based on the terra server that created the GCP notebook retrieved from
 # the VM metadata, if set.
-readonly TERRA_SERVER=$(get_metadata_value "instance/attributes/terra-cli-server")
+readonly TERRA_SERVER="$(get_metadata_value "instance/attributes/terra-cli-server")"
 if [[ -n "${TERRA_SERVER}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra server set --name=${TERRA_SERVER}"
 fi
 
 # Log in with app-default-credentials
 sudo -u "${JUPYTER_USER}" sh -c "terra auth login --mode=APP_DEFAULT_CREDENTIALS"
 
+####################################
+# Shell and notebook environment
+####################################
+
 # Set the CLI terra workspace id using the VM metadata, if set.
-readonly TERRA_WORKSPACE=$(get_metadata_value "instance/attributes/terra-workspace-id")
+readonly TERRA_WORKSPACE="$(get_metadata_value "instance/attributes/terra-workspace-id")"
 if [[ -n "${TERRA_WORKSPACE}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra workspace set --id=${TERRA_WORKSPACE}"
 fi
 
+# Set variables into the .bash_profile such that they are available
+# to terminals, notebooks, and other tools
+
+readonly GOOGLE_PROJECT="$(
```

Comment:
The TERRA env var which is used when running app command through TERRA has different variable naming from this. There we use GOOGLE_CLOUD_PROJECT, GOOGLE_SERVICE-ACCOUNT_EMAIL, (we don't have the equivalent of OWNER_EMAIL). I think we should at least keep the same naming. So the user doesn't get confused when the env is different when they are using terra CLI command vs not. 

As Melissa mentioned in the ticket, I think we should export all the TERRA env vars, not just these three. I propose we have a CLI command that returns all the env variable and we export them all to the bash_profile?

Reply:
Good point about name consistency.

To see env `terra cli` env vars, just run `terra app execute env`. I don't think we need a new command for this.

I think it's up to Matt if he wants to do the TERRA vars in this PR or not, it could always be a separate PR. :)

Reply:
Thanks Yu and Melissa. This is helpful additional context. For reference, in my notebook VM here are the relevant vars that I see:

```
(base) jupyter@<stuff>:~$ terra app execute env | sort | grep ^TERRA
TERRA_notebook_vm=projects/terra-<stuff>/locations/us-central1-a/instances/<stuff>
TERRA_vm_startup_test_bucket=gs://vm-startup-test-bucket

(base) jupyter@<stuff>:~$ terra app execute env | sort | grep ^GOOGLE
GOOGLE_CLOUD_PROJECT=terra-<stuff>
GOOGLE_PROJECT=terra-<stuff>
GOOGLE_SERVICE_ACCOUNT_EMAIL=pet-<stuff>@terra-<stuff>.iam.gserviceaccount.com
```

So addressing these as two separate variables - the TERRA variables are workspace resource references, while the GOOGLE/EMAIL variables are really Cloud-specific workspace properties.

I think exporting the TERRA variables is *likely* a nice feature to add, but I'm not sure. I would say that many notebooks would otherwise start with a series of "terra resolve" commands. But thinking this through a little more, if I'm working on a notebook, and I add a new resource, I'd like to be able to pick it up in my notebook right away. I could then use a "terra resolve" command in my notebook, but then my notebook reflects the order I was building things (sometimes I use the TERRA variable and sometimes I use the "terra resolve" command).

I want to put that aside for a moment. The specific implementation there is not obvious to me.

For the GOOGLE/EMAIL variables, things that do not change after the VM is started, and so setting the variables in the environment makes a lot of sense. It seems we have a few options for the notebook environments:

1. Keep the variable names [consistent with Leonardo](https://github.com/DataBiosphere/leonardo/blob/b1c19a89e374edcde524500e185e95080d22b657/http/src/main/resources/init-resources/startup.sh#L47)
2. Rationalize the variable naming as though Leonardo doesn't exist
3. Add both a new set of variables and support *key* Leonardo variables

I would lean towards (3) and keep a conservative eye on *key* Leonardo variables; I don't think it is a long list.

I think that the `GOOGLE_PROJECT` variable is very commonly used in app.terra.bio notebooks (any use of BigQuery, requester pays buckets, calls to `nextflow` and `dsub`, etc.)
The `OWNER_EMAIL` and `PET_SA_EMAIL` are used less frequently. The `OWNER_EMAIL` is an interesting one because it is really more of the "Terra user email" that happens today to be a Google account. The `PET_SA_EMAIL` is important functionally for things like `nextflow` and `dsub`.

I've implemented consistent with this. PTAL.

Reply:
I think I prefer 1. 3 works too.

Can we keep the terra cli env vars in sync with these env vars? This is where terra cli env are set:

https://cs.github.com/DataBiosphere/terra-cli/blob/70a0f6bffe2a00be51aa43f12c54631951e05b9c/src/main/java/bio/terra/cli/app/CommandRunner.java?q=GOOGLE_CLOUD_PROJECT#L89

Would you be up for doing the CLI PR too? :)

Reply:
Also, can you add a comment

```
# Keep in sync with terra CLI environment variables:
https://github.com/DataBiosphere/terra-cli/blob/main/src/main/java/bio/terra/cli/app/CommandRunner.java#L88
```

Reply:
Added.

Reply:
I'm happy to do the CLI PR too.
I wasn't able to get the code to build, but will follow-up with you offline.